### PR TITLE
DEV: Improvements to auto grid images

### PR DESCRIPTION
**Pull request by keegangeorge**
## 🔍 Overview
This PR is a follow-up to https://github.com/discourse/discourse/commit/ea1473e53260e8a458f8118b89e7b07b5cc50507. When we initially added the experimental feature for automatically adding `[grid]` to images, we add the [grid] surrounding images after all the uploads have been completed.

This can lead to confusion when `[grid]` is delayed to be added in the composer, as users may try to add grid manually leading to breakage. This also leads to issues with Discourse AI's automatic image caption feature.

**In this PR**: we simply move the logic to be added when the images are uploaded and processing. This way, `[grid]` surrounding images is added immediately. We also apply a fix for an edge-case to prevent images from being wrapped in `[grid]` when they are already inside `[grid]` tags.

No new tests added for auto grid image upload ordering, but new test case added for the edge-case fix.

## 🎥 Preview
https://github.com/user-attachments/assets/89f4b62c-426b-4b8f-a494-8f2ea1f40eb8

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/uppy/composer-upload.js [hunk 1]
```diff
@@ -735,20 +733,35 @@ export default class UppyComposerUpload {
   #autoGridImages() {
     const reply = this.composerModel.get("reply");
     const imagesToWrapGrid = new Set(this.#consecutiveImages);
-    const matches = reply.match(IMAGE_MARKDOWN_REGEX) || [];
+    const uploadingImagePattern = /\[Uploading: ([^\]]+?)\.\w+…\]\(\)/g;
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/lib/uppy/composer-upload.js:736]: Does this work with non-English locales? Not sure what we output here for `Processing:` or `Uploading:`.

**Review by pmusaraj — APPROVED** [review]
Just a comment on non-English locales, otherwise this looks good!
